invenio_app_rdm/records_ui/views/decorators.py

```diff
@@ -365,20 +367,169 @@ def view(**kwargs):
     return view
 
 
-def add_signposting(f):
-    """Add signposting link to view's response headers."""
+def _get_header(rel, value, link_type=None):
+    header = f'<{value}> ; rel="{rel}"'
+    if link_type:
+        header += f' ; type="{link_type}"'
+    return header
+
+
+def _get_signposting_cite_as(record):
+    """Release self url points to RDM record.
+
+    It points to DataCite URL if the integration is enabled, otherwise it points to the HTML URL.
+    """
+    doi_url = record["links"].get("doi")
+    html_url = record["links"]["self_html"]
+    return _get_header("cite-as", doi_url or html_url)
+
+
+def _get_signposting_types(record):
+    resource_type = record["metadata"]["resource_type"]
+    props = get_vocabulary_props(
+        "resourcetypes",
+        [
+            "props.schema.org",
+        ],
+        resource_type["id"],
+    )
+    url_schema_org = props.get("schema.org")
```

Comment:
Not sure if there's a better way to do this lookup.
I followed what's done in [`invenio_rdm_records/resources/serializers/signposting/schema.py`](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/resources/serializers/signposting/schema.py#L123-L142).

Reply:
Perhaps just check that it is cached so we don't query db on every landing page request

Reply:
From what I see these are indeed [cached here](https://github.com/inveniosoftware/invenio-vocabularies/blob/ca6e3ae71035cdfe785630fb6648bfa758d048fe/invenio_vocabularies/services/service.py#L120-L122), which is also mentioned [in `get_vocabulary_props`](https://github.com/inveniosoftware/invenio-rdm-records/blob/7e17fad16868ed40c2b65082246dcac77ad45fb1/invenio_rdm_records/resources/serializers/utils.py#L20-L24).